crud/src/main/java/org/openmbee/sdvc/crud/controllers/projects/ProjectsController.java

```diff
@@ -63,13 +66,15 @@ public ProjectsController(ProjectRepository projectRepository) {
         @RequestBody ProjectsRequest projectsPost) {
 
         if (projectsPost.getProjects().isEmpty()) {
-            ProjectsResponse pr = new ProjectsResponse();
-            pr.addMessage("empty");
-            return ResponseEntity.badRequest().body(pr);
+            throw new BadRequestException(new ProjectsResponse().addMessage("No projects"));
         }
+
         ProjectsResponse response = new ProjectsResponse();
         for (ProjectJson json: projectsPost.getProjects()) {
-            //TODO reject if projectId isn't there
+            if (json.getProjectId().isEmpty()) {
+                response.addMessage("Project ID is missing");
+                throw new BadRequestException(response);
```

Comment:
inside loop, response should include succeeded ones and failed ones, each service create and update may also throw exceptions so should catch them here to add to failed

Reply:
This is like the one above. The error message that we can send is not unique.

Reply:
i meant since it's inside a loop, if there's other project dbs already created and it throws an exception on a failed one, the caller would just see the whole thing as failed - that's ok if the transactional annotation actually deletes any created dbs